lib/strategies/webapp-strategy.js

```diff
@@ -344,12 +345,20 @@ function getTokens(req, options, strategy) {
 		code: req.query.code
 	};
 
+	function cb(err, user, info) {
+		if (err) {
+			return strategy.fail;
+		} else {
+			return strategy.success(user || null, info);
+		}
+	}
 	retrieveTokens(formData, strategy).then(function (appIdAuthContext) {
 		// Save authorization context to HTTP session
 		req.session[WebAppStrategy.AUTH_CONTEXT] = appIdAuthContext;
+		let user = appIdAuthContext.identityTokenPayload.sub;
 		delete req.session[WebAppStrategy.STATE_PARAMETER];
 		logger.debug("completeAuthorizationFlow :: success", options);
-		strategy.success(appIdAuthContext.identityTokenPayload || null);
+		strategy._verify(user, cb);
```

Comment:
Along the same lines, I think we should probably send the whole ID token to the user rather than just the sub field because they might want some of the other info in it to decide whether to verify the user